mvvm/src/main/java/com/thefuntasty/mvvm/delegate/BindingFragmentDelegate.kt

```diff
@@ -0,0 +1,44 @@
+package com.thefuntasty.mvvm.delegate
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.fragment.app.Fragment
+import com.thefuntasty.mvvm.BaseViewModel
+import com.thefuntasty.mvvm.ViewState
+
+interface BindingFragmentDelegate<VS : ViewState, B : ViewDataBinding> {
+
+    val brViewVariableId: Int
+    val brViewModelVariableId: Int
+    val brViewStateVariableId: Int
+    var binding: B
+
+    fun initViewBinding(
+        fragment: Fragment,
+        viewModel: BaseViewModel<VS>,
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        layoutResId: Int
+    ): Pair<B, View> {
+        return setupBindingView(inflater, container, layoutResId) {
+            it.setVariable(brViewVariableId, this)
+            it.setVariable(brViewModelVariableId, viewModel)
+            it.setVariable(brViewStateVariableId, viewModel.viewState)
+            it.setLifecycleOwner(fragment)
+        }
+    }
+
+    private fun setupBindingView(
+        layoutInflater: LayoutInflater,
+        container: ViewGroup?,
+        layoutResId: Int, set: (B) -> Unit
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/4a02c207953ea3674887f7e1a945e99f6b2b8a76/mvvm/src/main/java/com/thefuntasty/mvvm/delegate/BindingFragmentDelegate.kt#L37"></span>Parameter should be on a separate line (unless all parameters can fit a single line)</td>
    </tr>
  </tbody>
</table>